packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js

```diff
@@ -226,4 +232,23 @@ describe('Marketplace page', () => {
     const notInstalledText = queryByText(notInstalledCard, /copy install command/i);
     expect(notInstalledText).toBeVisible();
   });
+
+  it('only show install button for providers already installed', async () => {
+    // Open providers tab
+    render(App);
+    const providersTab = screen.getByRole('tab', { name: /providers/i });
+    fireEvent.click(providersTab);
+
+    // Provider that's already installed
+    const alreadyInstalledName = screen.getByRole('heading', { name: /cloudinary/i });
```

Comment:
why is this step necessary? Couldn't you just get the card with `getByTestId`?

Reply:
I want to make sure it's the cloudinary card, therefore that it's the closest npm-package-card parent of the cloudinary text.

This approach avoids looping through all the matches of npm-package-cards, which is thought was a bit expensive

Reply:
Actually looping seems faster, I'll update this right now